src/bootstrap/bin/rustc.rs

```diff
@@ -38,7 +38,23 @@ use std::path::PathBuf;
 use std::process::{Command, ExitStatus};
 
 fn main() {
-    let args = env::args_os().skip(1).collect::<Vec<_>>();
+    let mut args = env::args_os().skip(1).collect::<Vec<_>>();
+
+    // Insert the metadata suffix
+    if let Ok(s) = env::var("RUSTC_METADATA_SUFFIX") {
+        for i in 1..args.len() {
+            // Dirty code for borrowing issues
```

Comment:
This comment doesn't seem too helpful? Could you instead reference the comment in `lib.rs`?